pkg/dependency/parser/java/jar/parse.go

```diff
@@ -439,7 +439,24 @@ func (m manifest) determineVersion() (string, error) {
 }
 
 func removePackageDuplicates(pkgs []ftypes.Package) []ftypes.Package {
-	return lo.UniqBy(pkgs, func(pkg ftypes.Package) string {
-		return fmt.Sprintf("%s::%s::%s", pkg.Name, pkg.Version, pkg.FilePath)
-	})
+	// name::filePath => versions
+	var uniq = make(map[string][]mavenversion.Version)
+	var uniqPkgs []ftypes.Package
+	for _, pkg := range pkgs {
+		uniqID := pkg.Name + "::" + pkg.FilePath
+		// err is always nil
+		// cf. https://github.com/masahiro331/go-mvn-version/blob/d3157d602a08806ad94464c443e0cef1370694a1/version.go#L20-L25
+		pkgVer, _ := mavenversion.NewVersion(pkg.Version)
+		savedVers, ok := uniq[uniqID]
+		if !ok || !slices.ContainsFunc(savedVers, func(v mavenversion.Version) bool {
+			// There are times when patch `0` is omitted.
+			// So we can't compare versions just as strings
+			// for example `2.17.0` and `2.17` must be equal
+			return v.Equal(pkgVer)
+		}) {
+			uniq[uniqID] = []mavenversion.Version{pkgVer}
```

Comment:
Don't we have to append the new version here?
```suggestion
			uniq[uniqID] = append(uniq[uniqID], pkgVer)
```

Reply:
right! thanks!

Reply:
We should add a test so we will not have regression.

Reply:
Added `pom.properties` files for that in `io.quarkus.gizmo.gizmo-1.1.jar` file.
New structure of `io.quarkus.gizmo.gizmo-1.1.jar`:
https://github.com/aquasecurity/trivy/blob/7a5018e1d943839b85ea9a7df6404f82f1aaad89/pkg/dependency/parser/java/jar/parse_test.go#L172-L193